test/testStatuspage.js

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+
+const assert = require('assert');
+const nock = require('nock');
+const shell = require('shelljs');
+const sinon = require('sinon');
+
+const Statuspage = require('../src/statuspage/cli');
+const { getTimedPromise } = require('./utils');
+
+function buildArgs({
+  cmd, auth, pageId, name, desc, group, silent,
+} = {}) {
+  const args = [];
+  if (cmd) args.push(cmd);
+  if (auth) args.push('--auth', auth);
+  if (pageId) args.push('--page_id', pageId);
+  if (name) args.push('--name', `"${name}"`);
+  if (desc) args.push('--description', `"${desc}"`);
+  if (group) args.push('--group', group);
+  if (silent) args.push('--silent');
+  return args;
+}
+
+async function run(opts = {}) {
+  return new Statuspage().run(buildArgs(opts));
+}
+
+async function runShell(opts = {}) {
+  return shell.exec(`node ./src/statuspage ${buildArgs(opts).join(' ')}`);
+}
+
+describe('Testing statuspage', () => {
+  let name;
+  const logger = console;
+
+  // defaults
+  const cmd = 'setup';
+  const auth = 'test-auth';
+  const pageId = 'test-page-id';
+  const namePrefix = 'Test Component ';
+  const group = 'Test Group';
+  const email = 'component+abcdef@notifications.statuspage.io';
+  const now = new Date().toISOString();
+  const testComp = {
+    id: '1234',
+    page_id: pageId,
+    created_at: now,
+    updated_at: now,
+    group: false,
+    description: 'This is a test component',
+    position: 0,
+    status: 'operational',
+    showcase: true,
+    only_show_if_degraded: false,
+    automation_email: email,
+  };
+  const testGroup = {
+    id: '0000',
+    name: group,
+    page_id: pageId,
+    created_at: now,
+    updated_at: now,
+    group: true,
+    description: 'This is a test group',
+    position: 0,
+  };
+
+  before(() => {
+    try {
+      sinon.spy(logger, 'log');
+      sinon.spy(logger, 'error');
+    } catch (e) {
+      // ignore, already wrapped
```

Comment:
Not nice ;)
Wouldn't a `sinon.restore()` in the after avoid the exception ? (I am not sure, just asking)

Reply:
Good point, will try

Reply:
Worked, thanks!